tests/integration/utils2.go

```diff
@@ -654,23 +655,38 @@ func restartNodes(
 		nodes[i] = n
 	}
 
+	// The index of the action after the last wait action before the current restart action.
+	// We wish to resume the wait clock from this point onwards.
+	waitGroupStartIndex := 0
+actionLoop:
+	for i := actionIndex; i >= 0; i-- {
+		switch testCase.Actions[i].(type) {
+		case WaitForSync:
```

Comment:
question: for all other cases (default case for example) doesn't matter?

Reply:
For all other cases, the value would be `0` - the start of the slice